programs/mango-v4/src/instructions/admin_perp_withdraw_fees.rs

```diff
@@ -6,10 +6,16 @@ use crate::{accounts_ix::*, group_seeds};
 pub fn admin_perp_withdraw_fees(ctx: Context<AdminPerpWithdrawFees>) -> Result<()> {
     let group = ctx.accounts.group.load()?;
     let mut perp_market = ctx.accounts.perp_market.load_mut()?;
+    let bank = ctx.accounts.bank.load()?;
 
     let group_seeds = group_seeds!(group);
     let fees = perp_market.fees_settled.floor().to_num::<u64>() - perp_market.fees_withdrawn;
-    let amount = fees.min(ctx.accounts.vault.amount);
+    let amount = fees.min(
+        ctx.accounts
+            .vault
+            .amount
+            .saturating_sub(bank.unlendable_deposits),
```

Comment:
ah this is scary, how can we avoid using vault amounts just as is in future

Reply:
I've pushed more changes to the ts side where it was using vaults